### ISSUE TITLE
Intermediate checkpoints outperforming final checkpoint?

### ISSUE THREAD

**Issue by deeplearningmaniac** [facebookresearch/dinov2#424]
Hi -- this isn't an issue with the code, but I'm encountering some difficulty in training these models that I figured you might be able to help with.

I am applying DINOv2 to a specialized domain (which I unfortunately can't describe in any more detail). We have ~500M images for training, and we have implemented 16 validation tasks, some with small datasets (<10k) and some with large datasets (>100k).

We are running into something strange where when we evaluate the model on the validation tasks, a checkpoint taken 50% through training is outperforming the final model checkpoint at the end of training. We have tried tuning the learning rate, and it has helped a little bit, but we really expect the final checkpoint to have the best performance and across all learning rates we have tried, it does not.

Have you encountered this ever? Is there some hyperparameter(s) other than learning rate that might help fix this issue? Is our assumption correct that if we train the model well, then the 100% checkpoint should outperform all previous checkpoints?

Any suggestions or feedback would be much appreciated. Thank you!

**Comment by qasfb** [comment]
how bad ? can you plot the loss and eval curves ?

**Comment by deeplearningmaniac** [comment]
The loss curves seem relatively normal, continuing to decrease throughout training, but here are plots of the results of 16 evaluation tasks. As you can see, very often the models at 50% do better than the models at 100%. These are ViT-S models. For some of the tasks (eg, the bottom middle two), the curves are extremely irregular, and so these might not be great tasks, but the aforementioned behavior seems to be present in many of the others.

![evals](https://github.com/facebookresearch/dinov2/assets/148477540/d943f3b3-660f-48de-a9ab-d6a43ee9b3b6)

**Comment by ValentinKoch** [comment]
I had the same experience, not too surprised. we have plots showing similar behaviour here: https://arxiv.org/abs/2401.04720
